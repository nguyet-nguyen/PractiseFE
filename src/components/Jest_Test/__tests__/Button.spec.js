import React from "react";
import render from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
const {act} = TestRenderer;
import Button from '../Button.js';

describe("Unit Test Component render", () => {
    it("Button component rendering",  () => {
        const button = render.create(
            <Button newName='Lan Anh' />,
        );
        let tree = button.toJSON();
        expect(tree).toMatchSnapshot();

        act(() => {
            /* fire events that update state */
            tree.props.onClick();
        })
        tree = button.toJSON();
        expect(tree).toMatchSnapshot();
    })
})