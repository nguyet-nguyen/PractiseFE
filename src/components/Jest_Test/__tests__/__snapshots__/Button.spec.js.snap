// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test Component render Button component rendering 1`] = `
<button
  onClick={[Function]}
>
  Click 
  Me
</button>
`;

exports[`Unit Test Component render Button component rendering 2`] = `
<button
  onClick={[Function]}
>
  Click 
  Lan Anh
</button>
`;
